test_run_log_current_job_when_stopping is flaky

I've seen this test fail, in an arbitrary manner.

```
========================================================================================== FAILURES ===========================================================================================
___________________________________________________________________________ test_run_log_current_job_when_stopping ____________________________________________________________________________
                                                                                                                                                                                               
app = <procrastinate.app.App object at 0x7f2240c14d30>, running_worker = <procrastinate.worker.Worker object at 0x7f2240a6be50>                                                                
caplog = <_pytest.logging.LogCaptureFixture object at 0x7f2240a6bca0>                                                                                                                          
                                                                                                                                                                                               
    async def test_run_log_current_job_when_stopping(app, running_worker, caplog):                                                                                                             
        caplog.set_level("DEBUG")                                                                                                                                                              
                                                                                                                                                                                               
        @app.task(queue="some_queue")                                                                                                                                                          
        async def t():                                                                                                                                                                         
            running_worker.stop()                                                                                                                                                              
                                                                                                                                                                                               
        await t.configure(lock="sher").defer_async()                                                                                                                                           
                                                                                                                                                                                               
        try:                                                                                                                                                                                   
            await asyncio.wait_for(running_worker.task, timeout=0.5)                                                                                                                           
        except asyncio.TimeoutError:                                                                                                                                                           
            pytest.fail("Failed to launch task within .5s")                                                                                                                                    
                                                                                                                                                                                               
        # We want to make sure that the log that names the current running task fired.                                                                                                         
>       assert {                                                                                                                                                                               
            "Stop requested",                                                                                                                                                                  
            "Waiting for job to finish: worker 0: tests.integration.test_worker.t[1]() "                                                                                                       
            "(started 0.000 s ago)",                                                                                                                                                           
        } <= set(r.message for r in caplog.records)                                                                                                                                            
E       AssertionError: assert {'Stop requested', 'Waiting for job to finish: worker 0: tests.integration.test_worker.t[1]() (started 0.000 s ago)'} <= {'About to defer job tests.integration.
test_worker.t[None]()', 'Acknowledged job completion tests.integration.test_wor...01 s', 'Loaded job info, about to start job tests.integration.test_worker.t[1]()', 'Registering queue some_qu
eue', ...}                                                                                                                                                                                     
E        +  where {'About to defer job tests.integration.test_worker.t[None]()', 'Acknowledged job completion tests.integration.test_wor...01 s', 'Loaded job info, about to start job tests.in
tegration.test_worker.t[1]()', 'Registering queue some_queue', ...} = set(<generator object test_run_log_current_job_when_stopping.<locals>.<genexpr> at 0x7f2240bb7350>)

tests/integration/test_worker.py:71: AssertionError
-------------------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------------------
```

--- a/tests/integration/test_worker.py
+++ b/tests/integration/test_worker.py
@@ -68,11 +68,12 @@ async def test_run_log_current_job_when_stopping(app, running_worker, caplog):
         pytest.fail("Failed to launch task within .5s")
 
     # We want to make sure that the log that names the current running task fired.
-    assert {
-        "Stop requested",
-        "Waiting for job to finish: worker 0: tests.integration.test_worker.t[1]() "
-        "(started 0.000 s ago)",
-    } <= set(r.message for r in caplog.records)
+    logs = " ".join(r.message for r in caplog.records)
+    assert "Stop requested" in logs
+    assert (
+        "Waiting for job to finish: worker 0: tests.integration.test_worker.t[1]()"
+        in logs
+    )
 
 
 async def test_run_no_listen_notify(app):

